api.go

```diff
@@ -323,13 +323,17 @@ func RecoverCluster(conf *Config, fsm FSM, logs LogStore, stable StableStore,
 			continue
 		}
 
+		start := time.Now()
 		err = fsm.Restore(source)
 		// Close the source after the restore has completed
 		source.Close()
 		if err != nil {
 			// Same here, skip and try the next one.
 			continue
 		}
+		// Update the gauge for the time it took to restore.
+		metrics.SetGauge([]string{"raft", "fsm", "lastRestoreTime"},
+			float32(time.Since(start).Milliseconds()))
```

Comment:
A couple thoughts here.

1. The `Time` suffix in the name here could imply that this is a timer metric, since we use `Time` in other timers. Maybe `lastRestoreElapsed` or something else would help identify this as a gauge?
2. Generally metrics will only be reported from a single place. If we're going to report this metric in a few places I think it would be good to put it into a function and call that function from all the locations. The repetition isn't really a problem, but I do think that a function acts as a good hint to the reader that this metric breaks the common pattern of only being reported from a single place.

Reply:
I went with `Duration` as a suffix and moved to a central func. Good call.